Adds apply_to tags to Command line tools page

yetanothertw opened the pull request:
This section doesn't have the usual deployment type badges at the top. Since they are command line tools against Elasticsearch, they're not available on Elastic Cloud Hosted/Serverless.

Fixes [#306](https://github.com/elastic/docs-content-internal/issues/306)

I've been debating how to use badges most efficiently, and here's the iteration I settled on:

* Only use the tag for the deployment type that these tools apply to: self-managed. 
* Also add the self-managed tag to all the child pages.
* Add a bit of context about these utilities in the page description.
* In an admonition, explain the availability of the tools in other deployments and serverless.(I think an admonition is preferable to unavailable badges as it adds more context, otherwise we end up with badge overload).

[Preview available here](https://docs-v3-preview.elastic.dev/elastic/elasticsearch/pull/134538/reference/elasticsearch/command-line-tools)


------

<img width="936" height="696" alt="image" src="https://github.com/user-attachments/assets/eb6a8984-239e-4f7e-8ea7-379e6879ba90" />

yetanothertw committed:
Adds apply_to tags to Command line tools page

This section doesn't have the usual deployment type badges at the top. Since they are command line tools against Elasticsearch, they're not available on Elastic Cloud Hosted/Serverless.

Fixes [#306](https://github.com/elastic/docs-content-internal/issues/306)

yetanothertw committed:
Adding tip + removing some of the tags

yetanothertw committed:
Specify all the deployments in which the tools are unavailable in the tip

yetanothertw committed:
Remove the stack tag

colleenmcginnis reviewed (commented):
This approach makes sense to me. 🙂 

It aligns with the [Cumulative docs guidelines](https://elastic.github.io/docs-builder/contribute/cumulative-docs/guidelines/#tips):

>**Use `unavailable` sparingly**: For example, if a page is only about Elastic Cloud Hosted, don't add a `serverless: unavailable` tag. Refer to [When to indicate something is NOT applicable](https://elastic.github.io/docs-builder/contribute/cumulative-docs/guidelines/#when-to-indicate-something-is-not-applicable) for specific guidance.
with deployment methods I don't see a problem with it. 

And I personally think including applicability-related information in prose is ok as long as you're not specifying version/lifecycle information.

For all the other pages under this one, I would probably just include `applies_to.deployment.self: ga` without the content in the `tip`.

yetanothertw committed:
Add feedback from Edu

yetanothertw committed:
Adding the self-managed tag to all child pages

yetanothertw committed:
Merge branch 'main' into add-apply-tags-tools

yetanothertw committed:
Merge branch 'main' into add-apply-tags-tools

eedugon reviewed (approved):
LGTM! I'd ask for another quick review, maybe by Shaina :)

yetanothertw committed:
Specify they're not inteded for us in ECK and ECE deployments

Also, I've simplified the sentence introducing the list of tools to avoid repetition.

eedugon commented on docs/reference/elasticsearch/command-line-tools/index.md:
```diff
@@ -1,11 +1,22 @@
+
+:::{important}
+The tools are intended for use in self-managed environments and are included with all {{es}} software packages and Docker images, but are not accessible in fully managed {{ecloud}} deployments such as {{ech}} or {{serverless-full}}.
+:::
```
@shainaraskas , the tools can be used within docker images, hence they are available in ECE or ECK systems for the admins.

They are not available on ECH though, as our users are not platform admins.

Why do we want to say that they are not available (for example in ECK) where in some use cases they could help **exactly in the same way** as in a self-managed environment?
